node/silkworm/stagedsync/stage_history_index.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+   Copyright 2022 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#pragma once
+#include <boost/functional/hash.hpp>
+
+#include <silkworm/stagedsync/common.hpp>
+
+namespace silkworm::stagedsync {
+
+class HistoryIndex : public IStage {
+  public:
+    explicit HistoryIndex(NodeSettings* node_settings) : IStage(db::stages::kHistoryIndexKey, node_settings){};
+    ~HistoryIndex() override = default;
+
+    StageResult forward(db::RWTxn& txn) final;
+    StageResult unwind(db::RWTxn& txn, BlockNum to) final;
+    StageResult prune(db::RWTxn& txn) final;
+    std::vector<std::string> get_log_progress() final;
+
+  private:
+    std::unique_ptr<etl::Collector> collector_{nullptr};
+
+    std::atomic_bool loading_{false};  // Whether we're in ETL loading phase
+    std::string current_source_;       // Current source of data
+    std::string current_target_;       // Current target of transformed data
+    std::string current_key_;          // Actual processing key
+
+    StageResult forward_impl(db::RWTxn& txn, BlockNum from, BlockNum to, bool storage);
+    StageResult unwind_impl(db::RWTxn& txn, BlockNum from, BlockNum to, bool storage);
+    StageResult prune_impl(db::RWTxn& txn, BlockNum threshold, BlockNum to, bool storage);
+
+    //! \brief Collects bitmaps of block numbers changes for each account within provided
+    //! changeset boundaries
+    void collect_bitmaps_from_changeset(db::RWTxn& txn, const db::MapConfig& source_config, BlockNum from, BlockNum to,
+                                        bool storage);
+
+    //! \brief Collects unique keys touched by changesets within provided boundaries
+    std::unordered_map<Bytes, bool, boost::hash<Bytes>> collect_unique_keys_from_changeset(
```

Comment:
Why `boost::hash`?

Reply:
Cause std::hash has no specialization supported for `basic_string<uint8_t>`

Reply:
Actually rethinking ... I inherited unordered_map from previous code but is better to use a map cause eventually keys are processed in order.

Reply:
Removed the need for boost::hash